src/services/errorLog.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Error Log
+ *
+ * @module errorLog
+ */
+
+import http from './http.js';
+
+const postErrorLog = message => {
+  let currentTime = new Date();
+  let data = {
+    LOG_MESSAGE: message,
+    LOG_TIME: currentTime,
+  };
+  return http.post('log/add', data);
+};
+
+const postErrorMessage = message => {
+  return http.post('log', message);
+};
+
+//Code modified from https://medium.com/creative-technology-concepts-code/detect-device-browser-and-version-using-javascript-8b511906745
+function matchItem(string, data) {
+  var i = 0,
+    regex,
+    match;
+
+  for (i = 0; i < data.length; i += 1) {
+    regex = new RegExp(data[i].value, 'i');
+    match = regex.test(string);
+    if (match) {
+      return {
+        name: data[i].name,
+      };
+    }
+  }
+  return { name: 'unknown' };
+}
+
+const parseNavigator = navigator => {
+  //Code modified from https://medium.com/creative-technology-concepts-code/detect-device-browser-and-version-using-javascript-8b511906745
+  var oses = [
+    { name: 'Windows Phone', value: 'Windows Phone' },
```

Comment:
is oses plural for os? Is there a clearer name for this array?

Reply:
That was the intent. Perhaps just naming it "os" would be clearer? Or I could use "OperatingSystems" ?